Library/Homebrew/compat/md5.rb

```diff
@@ -1,26 +1,35 @@
 class Formula
   def self.md5(val)
+    # Don't throw an error at load time, just install time.
     stable.md5(val)
   end
 end
 
 class SoftwareSpec
   def md5(val)
+    odie <<-EOS.undent
+      Using MD5 has been dropped for security reasons.
+      Please switch to SHA-256.
+    EOS
```

Comment:
Would raising an appropriate exception address this, as it would prevent the formula from being loaded, but won't instantly kill the `brew` process like `odie` does? Is this what you are suggesting to do instead of using `odie`?